src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1325,8 +1340,10 @@ public void evaluatePermissionChain(SecuredUUID uuid) throws NotFoundException {
     @Override
     public void updateCertificateKeys(UUID keyUuid, String publicKeyFingerprint) {
         for (Certificate certificate : certificateRepository.findByPublicKeyFingerprint(publicKeyFingerprint)) {
-            certificate.setKeyUuid(keyUuid);
-            certificateRepository.save(certificate);
+            if (!certificate.isArchived()) {
```

Comment:
Relations to archived certificates can be maintained to keep inventory items relations actual. Certificate could be also unarchived and then missing correct existing relation to key already in inventory.